Context/ContextInterface.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/*
+ * This file is part of the FOSRestBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\RestBundle\Context;
+
+/**
+ * Stores the serialization or deserialization context (groups, version, ...).
+ *
+ * @author Ener-Getick <egetick@gmail.com>
+ */
+interface ContextInterface
```

Comment:
I think we might want to make a simpler ContextInterface maybe leaving out things like version, maxdepth and groups. and that for these 3 things we create additional interfaces that extend from ContextInterface. this way we are more flexible for a possible 3rd serializer that might not provide these features. also afaik the core serializer does not yet support version or max depth (yet).